pkg/storage/unified/resource/client.go

```diff
@@ -102,6 +146,45 @@ func idTokenExtractor(ctx context.Context) (string, error) {
 	return "", fmt.Errorf("id-token not found")
 }
 
+func namespaceExtractor(ctx context.Context) (string, []string, error) {
```

Comment:
I'm wondering if the namespace checker might be redundant given that we're retrieving the namespace from the user's identity. Currently, it seems like a no-op: we set the user's namespace in the header client-side, then verify it server-side against the user's identity.

The only scenario where the namespace header seems truly useful is if the caller's namespace is '*'. In that case, specifying the desired namespace in the header could be useful.

Reply:
cc: @ryantxu @kalleep

Reply:
Discussed live. The agreement is to keep the namespace checkers even though now they don't have a real utility, because there are future scenarios in which they'll be helpful.